### PR TITLE
feature/upgrade-packages-07-09-2022

### PR DESCRIPTION
**Pull request by alexndlm**
-upgrade all packages
-update tests
-add PHPStan tests to CI

### CODE REVIEW

File: .github/workflows/php-ci.yml [hunk 1]
```diff
@@ -35,6 +35,31 @@ jobs:
...
+                  php-version: ${{ matrix.php }}
+                  coverage: none
+
+            - name: Get the cached Composer dependencies
+              uses: eonx-com/actions-cache@v1
+              with:
+                  path: vendor
+                  key: ${{ matrix.php }}-composer-${{ hashFiles('composer.lock') }}
+
+            - name: Run Rector tests
```
Comment (roman-eonx) [inline .github/workflows/php-ci.yml:60]: Please update the name.

File: tests/PHPStan/ThrowExceptionMessageRule/ThrowExceptionMessageRuleTest.php [hunk 2]
```diff
@@ -4,23 +4,21 @@
...
-final class ThrowExceptionMessageRuleTest extends AbstractServiceAwareRuleTestCase
+final class ThrowExceptionMessageRuleTest extends RuleTestCase
 {
     public function getRule(): Rule
     {
-        return $this->getRuleFromConfig(
-            ThrowExceptionMessageRule::class,
-            __DIR__ . '/config/configured_rule.neon'
-        );
+        return new ThrowExceptionMessageRule(Exception::class);
```
Comment (roman-eonx) [inline tests/PHPStan/ThrowExceptionMessageRule/ThrowExceptionMessageRuleTest.php:21]: Let's pass the second argument explicitly to make the tests more obvious.

**Review by roman-eonx — CHANGES_REQUESTED** [review]

File: .github/workflows/php-ci.yml (hunk 1 above)
Comment (alexndlm) [reply .github/workflows/php-ci.yml]: Fixed

File: tests/PHPStan/ThrowExceptionMessageRule/ThrowExceptionMessageRuleTest.php (hunk 2 above)
Comment (alexndlm) [reply tests/PHPStan/ThrowExceptionMessageRule/ThrowExceptionMessageRuleTest.php]: Fixed

**Review by roman-eonx — APPROVED** [review]

**Review by mvlad — APPROVED** [review]

**Pull request merged**